pkg/apis/core/validation/shoot.go

```diff
@@ -1484,17 +1484,26 @@ func validateKubeletConfigReserved(reserved *core.KubeletConfigReserved, fldPath
 	return allErrs
 }
 
+var reservedTaintKeys = sets.NewString(v1beta1constants.TaintNodeCriticalComponentsNotReady)
```

Comment:
Right now, it would only resolve after a restart of the gardener-resource-manager, because the controller only reacts on `Create` events for `Nodes` that still have our taint.
We could consider `Update` events for `Nodes` that add back the taint.

However, I don't think this is a real problem. Should we consider it in your opinion?

Reply:
Not for now - the scenario is quite artificial, I believe. If we see real problems when this PR was put into practice, we can still address it.